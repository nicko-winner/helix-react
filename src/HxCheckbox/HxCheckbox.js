import React from 'react';
import PropTypes from 'prop-types';

/**
 * <HxCheckbox /> wrapper for <hx-checkbox />
 *
 * @example docs https://rackerlabs.github.io/helix-ui/components/checkboxes/
 * @example element https://rackerlabs.github.io/helix-ui/elements/hx-checkbox/
 */

class HxCheckbox extends React.PureComponent {
  componentDidMount() {
    if (this.hxRef) {
      this.hxRef.addEventListener('change', this.props.onChange);
    }
  }

  componentWillUnmount() {
    if (this.hxRef) {
      this.hxRef.removeEventListener('change', this.props.onChange);
    }
  }

  setRef = (element) => { this.hxRef = element; };

  render() {
    const { onChange, ...hxProps } = this.props;
    return (<hx-checkbox {...hxProps} ref={this.setRef} />);
  }
}

HxCheckbox.propTypes = {
  onChange: PropTypes.func,
  checked: PropTypes.bool,
  disabled: PropTypes.bool,
  indeterminate: PropTypes.bool
};

HxCheckbox.DefaultProps = {
  onChange: () => null
};

export default HxCheckbox;
