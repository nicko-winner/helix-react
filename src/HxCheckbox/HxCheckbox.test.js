import HxCheckbox from './HxCheckbox';
import React from 'react';
import * as enzyme from 'enzyme';

const mount = (props = {}) => {
  return enzyme.mount(<HxCheckbox {...props} />);
};

describe('HxCheckbox', () => {
  it('renders correctly', () => {
    expect(mount().html()).toMatchSnapshot();
  });

  describe('Checked', () => {
    it('renders correctly', () => {
      expect(mount({checked: true}).html()).toContain('checked');
    });
  });

  describe('Disabled', () => {
    it('renders correctly', () => {
      expect(mount({disabled: true}).html()).toContain('disabled');
    });
  });

  it('props.onChange is called on WebComponent `change` event', () => {
    const onChange = jest.fn();
    mount({onChange})
      .instance()
      .hxRef
      .dispatchEvent(new Event('change'));

    expect(onChange.mock.calls.length).toBe(1);
  });

  it('componentWillUnmount removes all event listeners', () => {
    const instance = mount().instance();
    instance.hxRef.removeEventListener = jest.fn();
    instance.componentWillUnmount();
    const listenersRemoved = instance.hxRef.removeEventListener.mock.calls.map((event) => event[0]);
    expect(['change']).toEqual(expect.arrayContaining(listenersRemoved));
  });
});
