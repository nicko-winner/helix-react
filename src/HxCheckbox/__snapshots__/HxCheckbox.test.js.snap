// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HxCheckbox renders correctly 1`] = `"<hx-checkbox></hx-checkbox>"`;
